core/services/job_runner.go

```diff
@@ -267,6 +267,14 @@ func executeRun(run *models.JobRun, store *store.Store) error {
 
 	if run.Status.Finished() {
 		run.SetFinishedAt()
+		reward := run.Overrides.Amount
+		if reward != nil {
+			earn := models.NewLinkEarned(run.JobSpecID, run.Overrides.Amount)
+			if err := store.AddLinkEarned(&earn); err != nil {
```

Comment:
Just wanted to capture this question: can we unintentionally double capture amount earned? Although this is primarily for presentation, I think upserting or having unique constraints on run id or the like could help. This is a thought, not a requirement.

Reply:
I was thinking this wouldn't be the case, but good point; having added JobRunID column with a unique constraint should take care of this.